packages/microcosm/src/history.js

```diff
@@ -336,13 +332,15 @@ class History extends Emitter {
     while (size > this.limit && root.complete) {
       size -= 1
 
-      this._emit('remove', root.parent)
-
       if (root.next) {
         root = root.next
       }
     }
 
+    if (root.parent) {
+      this._emit('remove', root.parent.parent, root.parent)
```

Comment:
This feels a little goofy. When we archive, we need to remove everything before the parent. Maybe this should just be `this._emit('remove', null, root.parent)`